libchisel/src/checkfp.rs

```diff
@@ -0,0 +1,217 @@
+use parity_wasm::elements::{Instruction, Module};
+
+use super::{ModuleError, ModuleValidator};
+
+/// Struct on which ModuleValidator is implemented.
+pub struct CheckFloatingPointPresenceFunc {}
+
+impl CheckFloatingPointPresenceFunc {
+    pub fn new() -> Self {
+        CheckFloatingPointPresenceFunc {}
+    }
+}
+
+impl ModuleValidator for CheckFloatingPointPresenceFunc {
+    fn validate(&self, module: &Module) -> Result<bool, ModuleError> {
+        for function in module.code_section().unwrap().bodies() {
```

Comment:
Can you use `module.code_section()?.bodies()` to have error handling?

Reply:
unfortunatly `code_section()` returns an `Option<&CodeSection>` where `NoneError` is marked as experimental. wasm-chisel seem to compile without any nightly-only experimental API's (see https://doc.rust-lang.org/std/option/struct.NoneError.html). I will just implement a simple check,  if `codesection()` returns none, `validate(..)` will return, `Err(ModuleError::NotFound)`.